tests/library/page-close.spec.ts

```diff
@@ -0,0 +1,290 @@
+  const promise = page.waitForEvent('dialog');
+  page.evaluate(() => alert()).catch(() => {});
+  const dialog = await promise;
+  await page.close();
+  const e = await dialog.dismiss().catch(e => e);
+  expect(e.message).toContain('Target page, context or browser has been closed');
+});
+
+test('expect should not print timed out error message when page closes', async ({ page }) => {
+  await page.setContent('<div id=node>Text content</div>');
+  const [error] = await Promise.all([
+    expect(page.locator('div')).toHaveText('hey', { timeout: 100000 }).catch(e => e),
+    page.close(),
+  ]);
+  expect(stripAnsi(error.message)).toContain(`expect(locator).toHaveText(expected)`);
+  expect(stripAnsi(error.message)).not.toContain('Timed out');
+});
+
+test('addLocatorHandler should throw when page closes', async ({ page, server, isAndroid }) => {
+  test.fixme(isAndroid, 'GPU process crash: https://issues.chromium.org/issues/324909825');
+  await page.goto(server.PREFIX + '/input/handle-locator.html');
+
+  await page.addLocatorHandler(page.getByText('This interstitial covers the button'), async () => {
+    await page.close();
+  });
+
+  await page.locator('#aside').hover();
+  await page.evaluate(() => {
+    (window as any).clicked = 0;
+    (window as any).setupAnnoyingInterstitial('mouseover', 1);
+  });
+  const error = await page.locator('#target').click().catch(e => e);
+  expect(error.message).toContain(kTargetClosedErrorMessage);
+});
+
+test('should reject all promises when page is closed', async ({ page, isAndroid }) => {
+  test.fixme(isAndroid, '"Target crashed" instead of "Target closed"');
+
+  let error = null;
+  await Promise.all([
+    page.evaluate(() => new Promise(r => {})).catch(e => error = e),
+    page.close(),
+  ]);
+  expect(error.message).toContain(kTargetClosedErrorMessage);
+});
+
+test('should set the page close state', async ({ page }) => {
+  expect(page.isClosed()).toBe(false);
+  await page.close();
+  expect(page.isClosed()).toBe(true);
+});
+
+test('should pass page to close event', async ({ page, isAndroid }) => {
+  test.fixme(isAndroid);
+
+  const [closedPage] = await Promise.all([
+    page.waitForEvent('close'),
+    page.close()
+  ]);
+  expect(closedPage).toBe(page);
+});
+
+test('should terminate network waiters', async ({ page, server, isAndroid }) => {
+  test.fixme(isAndroid);
+
+  const results = await Promise.all([
+    page.waitForRequest(server.EMPTY_PAGE).catch(e => e),
+    page.waitForResponse(server.EMPTY_PAGE).catch(e => e),
+    page.close()
+  ]);
+  for (let i = 0; i < 2; i++) {
+    const message = results[i].message;
+    expect(message).toContain(kTargetClosedErrorMessage);
+    expect(message).not.toContain('Timeout');
+  }
+});
+
+test('should be callable twice', async ({ page }) => {
+  await Promise.all([
+    page.close(),
+    page.close(),
+  ]);
+  await page.close();
+});
+
+test('should fire load when expected', async ({ page }) => {
```

Comment:
Quite a few of these tests have nothing to do with `page.close()`.

Reply:
Oops. Fixed.